components/prism-bash.js

```diff
@@ -19,66 +29,168 @@
+			pattern: /(^|[\s;|&]|[<>]\()(?:apropos|apt|apt-get|aptitude|aspell|awk|basename|bash|bc|bg|bzip2|cal|cat|cd|cfdisk|chgrp|chmod|chown|chroot|chkconfig|cksum|clear|column|cmp|comm|cp|cron|crontab|csplit|curl|cut|date|dc|dd|ddrescue|df|diff|diff3|dig|dir|dircolors|dirname|dirs|dmesg|du|egrep|eject|env|ethtool|expand|expect|expr|fdformat|fdisk|fg|fgrep|file|find|fmt|fold|format|free|fsck|ftp|fuser|gawk|git|grep|groupadd|groupdel|groupmod|groups|gzip|head|hg|history|hostname|htop|iconv|id|ifconfig|ifdown|ifup|import|install|jobs|join|kill|killall|less|link|ln|locate|logname|look|lpc|lpr|lprint|lprintd|lprintq|lprm|ls|lsof|make|man|mkdir|mkfifo|mkisofs|mknod|more|most|mount|mtools|mtr|mv|mmv|nano|netstat|nice|nl|nohup|notify-send|npm|nslookup|open|op|passwd|paste|pathchk|ping|pkill|popd|pr|printcap|printenv|ps|pushd|pv|quota|quotacheck|quotactl|ram|rar|rcp|reboot|rename|renice|remsync|rev|rm|rmdir|rsync|screen|scp|sdiff|sed|seq|service|shellcheck|shuf|sftp|sh|shutdown|sleep|slocate|sort|split|ssh|stat|strace|su|sudo|sum|suspend|sync|tac|tail|tar|tee|time|timeout|touch|top|traceroute|tr|tsort|tty|umount|uname|unexpand|uniq|units|unrar|unshar|uptime|useradd|userdel|usermod|users|uuencode|uudecode|v|vdir|vi|vmstat|wait|watch|wc|wget|whereis|which|who|whoami|write|xargs|xdg-open|yes|zenity|zip|zsh)(?=$|[\)\s;|&])/,
+			lookbehind: true
+		},
+		keyword: {
+			pattern: /(^|[\s;|&]|[<>]\()(?:if|then|else|elif|fi|for|while|in|case|esac|function|select|do|done|until)(?=$|[\)\s;|&])/,
 			lookbehind: true
 		},
-		'keyword': {
-			pattern: /(^|[\s;|&])(?:let|:|\.|if|then|else|elif|fi|for|break|continue|while|in|case|function|select|do|done|until|echo|exit|return|set|declare)(?=$|[\s;|&])/,
+        // https://www.gnu.org/software/bash/manual/html_node/Shell-Builtin-Commands.html
+		builtin: {
+			pattern: /(^|[\s;|&]|[<>]\()(?:\.|:|break|cd|continue|eval|exec|exit|export|getopts|hash|pwd|readonly|return|shift|test|times|trap|umask|unset|alias|bind|builtin|caller|command|declare|echo|enable|help|let|local|logout|mapfile|printf|read|readarray|source|type|typeset|ulimit|unalias|set|shopt)(?=$|[\)\s;|&])/,
 			lookbehind: true
 		},
-		'boolean': {
-			pattern: /(^|[\s;|&])(?:true|false)(?=$|[\s;|&])/,
+		boolean: {
+			pattern: /(^|[\s;|&]|[<>]\()(?:true|false)(?=$|[\)\s;|&])/,
 			lookbehind: true
 		},
-		'operator': /&&?|\|\|?|==?|!=?|<<<?|>>|<=?|>=?|=~/,
-		'punctuation': /\$?\(\(?|\)\)?|\.\.|[{}[\];]/
+		'file-descriptor': {
+			pattern: /(^|\W)(?:&[0-9])(?=$|\W)/,
+			lookbehind: true,
+			alias: 'important'
+		},
+		operator: {
+			// Lots of redirections here, but not just that.
+			pattern: /\d?<>|>\||\+=|==?|!=?|=~|<<[<-]?|[&\d]?>>|\d?[<>]&?|&>|&&?|\|&|\|\|?|<=?|>=?/,
```

Comment:
I actually changed this regexp tons of times and noticed that the slightest change, even the most innocent one, could break multiple-character operators into several occurrences of shorter operators, like a `<>` being interpreted as `<` followed by `>` rather than as a single operator. So I have to admit I grew kind of scared of this part of the code. :fearful: I'll give it a try, I guess, but I think readability should (alas) become secondary for this particular line. But it's too soon to give up altogether.